### ISSUE TITLE
Error cell width  with the first table after page-break-after: avoid;

### ISSUE THREAD

**Issue by sergiocc96** [dompdf/dompdf#2353]
Hello i have a a table with `table-layout: fixed !important;` and `page-break-after: avoid;`  i want the second cell have 60% of the width. Its all ok except with the first table of every new page. Anybody knows what is the problem?
Thanks

![xOuph](https://user-images.githubusercontent.com/53588588/105579189-42578e80-5d85-11eb-891b-e9fe128a3e7a.png)

**Comment by bsweeney** [comment]
Do you have sample HTML we can test with? The current implementation of table-layout: fixed just evenly divides the width amongst the columns, disregarding the specified width (#1432).

**Comment by jellesmets** [comment]
@sergiocc96 
I had a similar issue as yours and circumvented this issue using the `colspan` attribute merging columns together to obtain a certain column width. The colspan attribute should be added to every row `th` and `td`.
E.g. to become a table with 1st column 30% width, and 2nd column 70% width
```
<table style="width:100%; table-layout: fixed;">
  <tr>
    <th colspan="3">Name</th>
    <th colspan="7">Telephone</th>
  </tr>
  <tr>
    <td colspan="3">Bill Gates</td>
    <td colspan="7">55577854</td>
  </tr>
</table>
```

This might not be satisfactory in all cases, but can help as a quickfix until the issue is resolved.

**Comment by jamieburchell** [comment]
Pulling my hair out with a possibly related issue where the table width declarations (either as CSS or width attributes) are completely ignored when the table is forced on a new page, but if it sits across page boundaries the widths are respected.

**Comment by jamieburchell** [comment]
Here's the minimal code required to produce the issue on an A4 portrait PDF:

`
<html>
<body>
<div style="height: 1010px"></div>
<table width="100%" cellpadding="5" cellspacing="0" border="0">
  <tr>
    <th width="50%">Additional extras at the time of enquiry:</th>
    <th width="10%">Units</th>
    <th width="20%">Unit cost</th>
    <th width="20%">Total</th>
  </tr>
  <tr>
    <td>Lorem, ipsum dolor sit amet consectetur, adipisicing elit. Similique aperiam autem esse obcaecati fuga dolor velit nihil exercitationem repudiandae odio, reiciendis quidem, quo laudantium nostrum, blanditiis ex enim non molestiae.</td>
    <td>2</td>
    <td>£1000</td>
    <td>£2000</td>
</tr>
</table></body>
</html>
`

The table widths are not respected unless the first div is removed, or its height tweaked.

**Comment by jamieburchell** [comment]
Here's the minimal code required to produce the issue on an A4 portrait PDF:

```
<html>
<body>
<div style="height: 1010px"></div>
<table width="100%" cellpadding="5" cellspacing="0" border="0">
  <tr>
    <th width="50%">Amet consectetur, adipisicing elit:</th>
    <th width="10%">Units</th>
    <th width="20%">Unit cost</th>
    <th width="20%">Total</th>
  </tr>
  <tr>
    <td>Lorem, ipsum dolor sit amet consectetur, adipisicing elit. Similique aperiam autem esse obcaecati fuga dolor velit nihil exercitationem repudiandae odio, reiciendis quidem, quo laudantium nostrum, blanditiis ex enim non molestiae.</td>
    <td>2</td>
    <td>£1000</td>
    <td>£2000</td>
</tr>
</table></body>
</html>
```

The table widths are not respected unless the first div is removed, or its height tweaked.

**Comment by gmeister2** [comment]
> jamieburchell
Pulling my hair out with a possibly related issue where the table width declarations (either as CSS or width attributes) are completely ignored when the table is forced on a new page, but if it sits across page boundaries the widths are respected.

Could above be related to this issue:
https://github.com/dompdf/dompdf/issues/2395

Is there information on when and where in the code this issue was introduced?
For my use case version 0.8.5 works fine. Version 1.0.2 does not.

Are there other solutions than adding a `colspan` attribute to each `th` and `td`?

**Comment by Mellthas** [comment]
This should be fixed by #2502, unless there is another issue at play here.
